src/scheduler/executionHistory.tsx

```diff
@@ -0,0 +1,219 @@
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { PickersDayProps, PickersDay } from '@mui/x-date-pickers/PickersDay';
+import { DateCalendar } from '@mui/x-date-pickers/DateCalendar';
+import dayjs, { Dayjs } from 'dayjs';
+import ListDagRuns from './listDagRuns';
+import { LabIcon } from '@jupyterlab/ui-components';
+import LeftArrowIcon from '../../style/icons/left_arrow_icon.svg';
+import ListDagTaskInstances from './listDagTaskInstances';
+
+const iconLeftArrow = new LabIcon({
+  name: 'launcher:left-arrow-icon',
+  svgstr: LeftArrowIcon
+});
+
+const ExecutionHistory = ({
+  composerName,
+  dagId,
+  handleBackButton
+}: {
+  composerName: string;
+  dagId: string;
+  handleBackButton: () => void;
+}): JSX.Element => {
+  const [dagRunId, setDagRunId] = useState('');
+  const currentDate = new Date().toLocaleDateString();
+  const [selectedDate, setSelectedDate] = useState<Dayjs | null>(null);
+  const [startDate, setStartDate] = useState('');
+  const [endDate, setEndDate] = useState('');
+
+  const [blueListDates, setBlueListDates] = useState<string[]>([]);
+  const [greyListDates, setGreyListDates] = useState<string[]>([]);
+  const [orangeListDates, setOrangeListDates] = useState<string[]>([]);
+  const [redListDates, setRedListDates] = useState<string[]>([]);
+  const [greenListDates, setGreenListDates] = useState<string[]>([]);
+  const [darkGreenListDates, setDarkGreenListDates] = useState<string[]>([]);
+
+  const handleMonthChange = () => {
+    setDagRunId('');
+    setSelectedDate(null);
+  };
+
+  const handleDateSelection = (selectedValue: any) => {
+    setDagRunId('');
+    setSelectedDate(selectedValue);
+  };
+
+  const CustomDay = (props: PickersDayProps<Dayjs>) => {
+    const { day, isFirstVisibleCell, isLastVisibleCell } = props;
+    if (isFirstVisibleCell) {
+      setStartDate(new Date(day.toDate()).toISOString());
+    }
+    if (isLastVisibleCell) {
+      const nextDate = new Date(day.toDate());
+      nextDate.setDate(day.toDate().getDate() + 1);
+      setEndDate(nextDate.toISOString());
+    }
+
+    const totalViewDates = day.date();
+
+    //Color codes to highlight dates in calendar 
+    //Blue color code for running status
+    const isBlueExecution =
+      blueListDates.length > 0 &&
+      blueListDates.includes(totalViewDates.toString().padStart(2, '0'));
```

Comment:
move totalViewDates.toString().padStart(2, '0') into a constant and reuse it?  Why do you need to padStart?

Reply:
Resolved - Moved to constant and needed for Mui calendar expecting 2 digit day format.